As of HHVM 3.15 pgsql is supported, add php 7.1 to tests

photodude opened the pull request:
- Remove the HHVM exclusion now that pgsql is in core HHVM
- add PHP 7.1
- fix the xdebug line that was erroring the php 7.1+ and HHVM builds

photodude committed:
As of HHVM 3.15 pgsql is supported

photodude committed:
pin to HHVM 3.15 and add PHP 7.1

Stop testing against old HHVM 3.9 and test against current LTS 3.15

photodude committed:
fixing the xdebug error with php 7.1, nightly and HHVM-3.15

photodude committed:
Fast finish to avoid waiting on allowed failures

Also Allow HHVM 3.15 with pgsql to fail

photodude committed:
Set all HHVM-3.15 to allowed failure

photodude commented:
@Ocramius this is ready for consideration now.

Ocramius commented on .travis.yml:
```diff
@@ -26,7 +26,34 @@ after_script:
 matrix:
+  fast_finish: true
   include:
+    - php: hhvm-3.15
```
This is painful to maintain, and will require changing every time there is a minor version bump. Can we just rely on `hhvm` at the base config, and have travis-ci update to latest stable?

photodude replied:
Yes and no...

I chose [3.15 since it's LTS](http://hhvm.com/blog/6083/hhvm-long-term-support), I believe the next LTS will be 3.18 or 3.20, or maybe 4.0. Considering 3.15 like an LTS php version 5.6 or 7.0

To be on the actual latest of HHVM we can skip specifying the version, but we can't put it in the base configuration since the Trusty container builds on Travis CI have not been released, and the precise containers only support HHVM 3.9. [See HHVM options](https://docs.travis-ci.com/user/languages/php#HHVM-versions)

So it's just a matter of do you only want to test latest, LTS versions, both, etc. 

We can also add builds for [HHVM in php 7 mode](https://docs.hhvm.com/hhvm/configuration/INI-settings#php-7-settings) rather than the default php 5 mode. (I skipped that option since [HHVM has an issue with the PHP 7 mode](https://github.com/facebook/hhvm/issues/7198) that is blocking composer from working)

Ocramius commented on .travis.yml:
```diff
@@ -35,10 +62,17 @@ matrix:
       env: DB=mariadb
       addons:
         mariadb: 5.5
+    - php: hhvm-3.15
```
Same as above

Ocramius commented on .travis.yml:
```diff
@@ -48,15 +82,21 @@ matrix:
-  exclude:
-    - php: hhvm
-      env: DB=pgsql  # driver for PostgreSQL currently unsupported by HHVM, requires 3rd party dependency
+    - php: hhvm-3.15
```
Same

photodude committed:
Set to HHVM latest (currently 3.15 until next release)

Move sudo: false to the top line to be more clear about use of containers as default vs the trusty builds for HHVM

photodude committed:
Remove the last hhvm3.15 tag

photodude commented:
@Ocramius can the marina 5.5 builds be removed?

Ocramius commented:
@photodude yes, we can safely move everything to 5.6+

photodude committed:
Drop mariadb: 5.5 tests

reorder matrix so all HHVM tests are at the bottom

photodude commented:
Done, ready for review again.

photodude commented:
@Ocramius bump for merge consideration

Ocramius commented:
@photodude 👍 thanks for the poke!

photodude commented:
You are welcome
